### PR TITLE
Add multiple init pods to scheduler perf test cases

### PR DESCRIPTION
**Pull request by alculquicondor**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Allowing different types of init pods we can represent more realistic test cases for benchmarks and profiling.

I'm adding a test case where there are several init pods with affinity or antiaffinity, and we measure basic pods.

I'm also surfacing suppressed errors and replacing `klog.Fatalf` with `b.Fatalf`. The previous approach wasn't playing well with the test tool.

Sample run: https://gist.github.com/alculquicondor/5d98a2f6534a8fd6dca8564c3d75b196

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/integration/scheduler_perf/scheduler_perf_test.go [hunk 1]
```diff
@@ -200,91 +223,119 @@ func getNodePreparer(nc nodeCase, clientset clientset.Interface) testutils.TestN
...
+		return nil, fmt.Errorf("parsing test cases: %v", err)
+	}
 
 	testCases := make([]testCase, 0)
 	for _, s := range simpleTests {
 		testCase := s.Template
 		for _, p := range s.Params {
 			testCase.Nodes.Num = p.NumNodes
-			testCase.InitPods.Num = p.NumInitPods
+			testCase.InitPods = append([]podCase(nil), testCase.InitPods...)
```
Comment (ingvagabund) [inline test/integration/scheduler_perf/scheduler_perf_test.go]: Why is this change necessary?

**Review by ingvagabund — CHANGES_REQUESTED** [review]
Few nits though looks good in overall. Can you also split the PR into two commits? First one where the functions return error and the second one that adds multiple init pods. So it's easier to follow the changes in the future.

File: test/integration/scheduler_perf/scheduler_perf_test.go [hunk 2]
```diff
@@ -165,18 +187,19 @@ func perfScheduling(test testCase, b *testing.B) []DataItem {
 	return dataItems
 }
 
-func waitNumPodsScheduled(num int, podInformer coreinformers.PodInformer) {
+func waitNumPodsScheduled(b *testing.B, num int, podInformer coreinformers.PodInformer) error {
 	for {
 		scheduled, err := getScheduledPods(podInformer)
 		if err != nil {
-			klog.Fatalf("%v", err)
+			return nil
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply test/integration/scheduler_perf/scheduler_perf_test.go]: Oops. Fixed.

File: test/integration/scheduler_perf/scheduler_perf_test.go (hunk 1 above)
Comment (alculquicondor) [reply test/integration/scheduler_perf/scheduler_perf_test.go]: Otherwise the slice is shared and a second iteration modifies both test cases.

**Comment by alculquicondor** [comment]
> Can you also split the PR into two commits?

That would require to rewrite a lot of the PR. I don't think the "return error" changes are significant. Is that fine with you?

**Comment by ingvagabund** [comment]
> That would require to rewrite a lot of the PR. I don't think the "return error" changes are significant. Is that fine with you?

1. checkout master
2. copy paste `scheduler_perf_test.go` from your PR
3. undo changes adding multiple init pods (should not be much)
4. commits changes
5. copy paste all the files from your PR

It's not a must though it will help to make the git history more friendly. Please, reconsider.

**Comment by alculquicondor** [comment]
Done. Thanks for reminding me to be mindful of traceability.

**Comment by alculquicondor** [comment]
ping @ingvagabund

**Comment by ingvagabund** [comment]
Sorry, I got distracted, to many tabs open.

/lgtm
